internal/common/voting/VotingMethods.go

```diff
@@ -0,0 +1,304 @@
+package voting
+
+import (
+	"math"
+	"sort"
+
+	"github.com/google/uuid"
+)
+
+// Auxiliary Structure for Sorting
+type kv struct {
+	Key   uuid.UUID
+	Value float64
+}
+
+func Plurality(voteList []map[uuid.UUID]float64) uuid.UUID {
+	/*
+		Plurality:
+			Each voter selects one candidate and the candidate with the most first-placed votes is the winner.
+	*/
+	voteCount := make(map[uuid.UUID]float64)
+	var winner uuid.UUID
+
+	for _, preference := range voteList {
+		var maxPreference float64
+		var firstLootBoxChoice uuid.UUID
+		for lootBox, key := range preference {
+			if key > maxPreference {
+				firstLootBoxChoice = lootBox
+			}
+		}
+		voteCount[firstLootBoxChoice]++
+	}
+
+	// final step: we need to find the winner with highest count number in map.
+	var maxVotes float64
+
+	for lootBox, votes := range voteCount {
+		if votes > maxVotes {
+			maxVotes = votes
+			winner = lootBox
+		}
+	}
+
+	// return the final winner
+	return winner
+}
+
+func Runoff(voteList []map[uuid.UUID]float64) uuid.UUID {
```

Comment:
I could well have misinterpretted this voting method, but doesn't this require the agents to revote for round two?